commpy/links.py

```diff
@@ -0,0 +1,148 @@
+   linkModel            -- Link model object.
+"""
+from __future__ import division  # Python 2 compatibility
+
+import numpy as np
+
+from commpy.channels import MIMOFlatChannel
+
+__all__ = ['link_performance', 'linkModel']
+
+
+def link_performance(link_model, SNRs, send_max, err_min, send_chunk=None, code_rate=1):
+    """
+    Estimate the BER performance of a link model with Monte Carlo simulation.
+
+    Parameters
+    ----------
+    link_model : linkModel object.
+
+    SNRs : 1D arraylike
+           Signal to Noise ratio in dB defined as :math:`SNR_{dB} = (E_b/N_0)_{dB} + 10 \log_{10}(R_cM_c)`
+           where :math:`Rc` is the code rate and :math:`Mc` the modulation rate.
+
+    send_max : int
+               Maximum number of bits send for each SNR.
+
+    err_min : int
+              link_performance send bits until it reach err_min errors (see also send_max).
+
+    send_chunk : int
+                  Number of bits to be send at each iteration.
+                  *Default*: send_chunck = err_min
+
+    code_rate : float in (0,1]
+                Rate of the used code.
+                *Default*: 1 i.e. no code.
+
+    Returns
+    -------
+    BERs : 1d ndarray
+           Estimated Bit Error Ratio corresponding to each SNRs
+    """
+
+    # Initialization
+    BERs = np.empty_like(SNRs, dtype=float)
+    # Set chunk size and round it to be a multiple of num_bits_symbol*nb_tx to avoid padding
+    if send_chunk is None:
+        send_chunk = err_min
+    divider = link_model.num_bits_symbol * link_model.channel.nb_tx
+    send_chunk = max(divider, send_chunk // divider * divider)
+
+    # Computations
+    for id_SNR in range(len(SNRs)):
+        link_model.channel.set_SNR_dB(SNRs[id_SNR], code_rate, link_model.Es)
+        bit_send = 0
+        bit_err = 0
+        receive_size = link_model.channel.nb_tx * link_model.num_bits_symbol
+        while bit_send < send_max and bit_err < err_min:
+            # Propagate some bits
+            msg = np.random.choice((0, 1), send_chunk)
+            symbs = link_model.modulate(msg)
+            channel_output = link_model.channel.propagate(symbs)
+
+            # Deals with MIMO channel
+            if isinstance(link_model.channel, MIMOFlatChannel):
+                nb_symb_vector = len(channel_output)
+                received_msg = np.empty_like(msg, int)
+                for i in range(nb_symb_vector):
+                     received_msg[receive_size * i:receive_size * (i+1)] = \
+                         link_model.receive(channel_output[i], link_model.channel.channel_gains[i], link_model.constellation)
+            else:
+                received_msg = link_model.receive(channel_output, link_model.channel.channel_gains, link_model.constellation)
+            # Count errors
+            bit_err += (msg != received_msg).sum()  # Remove MIMO padding
+            bit_send += send_chunk
+        BERs[id_SNR] = bit_err / bit_send
+    return BERs
+
+
+class linkModel:
```

Comment:
@BastienTr @akou97 Most of the PR looks good. Thank you! Small request on the naming conventions. Can we stick to using PascalCase / UpperCamelCase for class names e.g., linkModel would become "LinkModel" 
and snake_case for function (including member functions) names and variables.